### PR TITLE
SIGINT-1028: Support for new FixPR capability for Black Duck Hub

### PR DESCRIPTION
**Pull request by spurohitsynopsys**

### CODE REVIEW

File: synopsys-task/src/synopsys-task/tools-parameter.ts [hunk 1]
```diff
@@ -174,13 +175,13 @@ export class SynopsysToolsParameter {
...
 
     // Check and put environment variable for fix pull request
-    if (parseToBoolean(inputs.BLACKDUCK_AUTOMATION_FIXPR_KEY)) {
-      console.log("Blackduck Automation Fix PR is enabled");
-      blackduckData.data.azure = this.getAzureRepoInfo();
-      blackduckData.data.blackduck.automation.fixpr = true;
+    if (parseToBoolean(inputs.BLACKDUCK_FIXPR_ENABLED)) {
+      console.log("Black Duck Fix PR is enabled");
+      blackduckData.data.blackduck.fixpr = this.setBlackDuckFixPrInputs();
+      //blackduckData.data.azure = this.getAzureRepoInfo();
```
[the remark this answers is not in the document]
Comment (spurohitsynopsys) [reply synopsys-task/src/synopsys-task/tools-parameter.ts]: updated

**Review by lsynopsys — APPROVED** [review]

File: synopsys-task/src/synopsys-task/input.ts [hunk 2]
```diff
@@ -71,10 +71,29 @@ export const BLACKDUCK_SCAN_FAILURE_SEVERITIES =
...
+  taskLib.getInput(constants.BLACKDUCK_FIXPR_ENABLED_KEY)?.trim() ||
+  "";
 export const BLACKDUCK_AUTOMATION_PRCOMMENT =
   taskLib.getInput(constants.BLACKDUCK_AUTOMATION_PRCOMMENT_KEY) || "";
+export const BLACKDUCK_FIXPR_MAXCOUNT =
+  taskLib.getInput(constants.BLACKDUCK_FIXPR_MAXCOUNT_KEY)?.trim() || "";
+export const BLACKDUCK_FIXPR_CREATE_SINGLE_PR =
+  taskLib.getInput(constants.BLACKDUCK_FIXPR_CREATE_SINGLE_PR_KEY)?.trim() ||
+  "";
+export const BLACKDUCK_FIXPR_FILTER_BY =
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/input.ts:85]: we need to remove this input, not applicable now

File: synopsys-task/src/synopsys-task/tools-parameter.ts [hunk 3]
```diff
@@ -302,6 +303,62 @@ export class SynopsysToolsParameter {
...
+      inputs.BLACKDUCK_FIXPR_FILTER_SEVERITIES != null &&
+      inputs.BLACKDUCK_FIXPR_FILTER_SEVERITIES.length > 0
+    ) {
+      for (const fixPrSeverity of inputs.BLACKDUCK_FIXPR_FILTER_SEVERITIES) {
+        if (fixPrSeverity != null && fixPrSeverity !== "") {
+          fixPRFilterSeverities.push(fixPrSeverity.trim());
+        }
+      }
+      blackDuckFixPrData.filter = {
+        ...(inputs.BLACKDUCK_FIXPR_FILTER_BY
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/tools-parameter.ts:351]: we need to remove BLACKDUCK_FIXPR_FILTER_BY, not applicable now

Comment (lsynopsys) [reply synopsys-task/src/synopsys-task/tools-parameter.ts]: done

**Pull request closed without merging**